src/screens/QuestionScreen/QuestionScreen.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import { View, Text, Image, TouchableOpacity, SafeAreaView, TextInput, Alert, Linking } from 'react-native';
+import { firebase } from '../../config/config';
+import { LOGO } from '../../config/styles.js';
+import styles from './styles';
+import moment from 'moment';
+import Icon from 'react-native-vector-icons/Ionicons';
+import ImagePicker from 'react-native-image-crop-picker';
+import * as Progress from 'react-native-progress';
+
+class QuestionScreen extends  React.Component {
+    state = {QueryInput : '', setimage : null, setDisplayImage: null, transferred: 0};
+
+    takephotofrommlib = () => {
+        const {setimage} = this.state
+        this.setState({transferred: 0})
+        ImagePicker.openPicker({
+          width: 300,
+          height: 300,
+          cropping: true,
+        }).then((image) => {
+          const imageUri = image.path;
+          this.setState({setimage: imageUri});
+        }).catch("Unknown Error Occured")
+      };
+
+      uploadImage = async () => {
+        const {setimage} = this.state
+        if(setimage == null){
+          return
+        }
+          const uri = setimage;
+          const childPath = `post/${firebase.auth().currentUser.uid}//${Math.random().toString(36)}`;
```

Comment:
Sorry, I missed what I meant here. I have a question, when we give answerers, are they also posted into the same object that is created in the post/uid URI? Change this back to original what you had earlier.

Reply:
No! When a user gives an answer, A new collection is created in that particular Query Collection. But the image is saved in firebase storage, whose URL is saved in this answer collection. `childPath` is the URI of that image in firebase storage and `uri` is the image location in local storage

<img width="1260" alt="Screenshot 2021-06-19 at 3 06 47 PM" src="https://user-images.githubusercontent.com/40908684/122638143-211a8480-d110-11eb-97b2-c56fb79b7401.png">

Reply:
Every image when posted will be saved in a location with the format of childPath. So every image posted by a particular user in this application will be saved in a folder with his uid. Here is the path of images in firebase storage in below screenshot

<img width="1135" alt="Screenshot 2021-06-19 at 3 17 11 PM" src="https://user-images.githubusercontent.com/40908684/122638437-b23e2b00-d111-11eb-8adc-0921e38e19d1.png">

Reply:
Oh OK, in that case shall we change the URL to post/uid/queries/random_string. That would organize the files more convenient way I think. Better if you can organize the images/file based by queries. Let me know what you are going to do here then.

Reply:
+1 @shehand. Better to organize the files in a better way.

Reply:
Yes! I can change the URL to `post/uid/queries/random_string`, but changing it to `post/uid/queries/post_id/random_string` is not possible, because at first, I upload the image in firebase storage, and then we post the query. At the time of uploading of the image, we don't have post_id. It creates when we post the query. Let me know if changing it to `post/uid/queries/random_string` is fine.

Reply:
That's where you come up with a creative approach. You can do it when a variable is declared for each user which hold the next valid post id. This value is incremented by one each time new post is posted by the user. There can be situation when providing answers, need to add images. In that case we can add those kind of images to the same folder and this will make the storage kind of organized(Note: nothing is impossible, you just have to think out of the box). Let us know you got the idea what we are referring here.

Reply:
Okay! I Got it. Let me Brainstorm the idea and come up with a good solution

Reply:
Here is the Screenshot of the path of the image.

<img width="1143" alt="Screenshot 2021-06-20 at 7 43 52 PM" src="https://user-images.githubusercontent.com/40908684/122677506-31119180-d200-11eb-8380-2fb8c3057e82.png">


What I did is- I created a new field in the user section `postNumber` which will be incremented by one every time a user post a query. So the path is `post/uid/queries/postNumber/randomString`. `postNumber` field can also be used as a post count by an individual user.

@shehand I guess this is what you wanted me to implement!? Let me know if this is fine.